airsonic-main/src/main/java/org/airsonic/player/service/MediaScannerService.java

```diff
@@ -228,38 +232,44 @@ private void doScanLibrary() {
     }
 
     private void scanFile(MediaFile file, MusicFolder musicFolder, Date lastScanned,
-                          Map<String, Integer> albumCount, Genres genres, boolean isPodcast) {
+                          Map<String, Integer> albumCount, Map<String, Genre> genreCache, boolean isPodcast) {
         scanCount++;
         if (scanCount % 250 == 0) {
             LOG.info("Scanned media library with " + scanCount + " entries.");
         }
 
         LOG.trace("Scanning file {}", file.getPath());
 
-        searchService.index(file);
-
         // Update the root folder if it has changed.
         if (!musicFolder.getPath().getPath().equals(file.getFolder())) {
             file.setFolder(musicFolder.getPath().getPath());
             mediaFileDao.createOrUpdateMediaFile(file);
         }
 
         if (file.isDirectory()) {
+            searchService.index(file, Collections.emptyList());
```

Comment:
I see, this is because it has been removed previously. But you could simplify that by not having to treat differently the isDirectory case. See my other comment bellow.